upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -222,9 +222,13 @@ func (c *ApplyClusterCmd) Run(ctx context.Context) error {
 		return err
 	}
 
-	err = c.validateKubernetesVersion()
-	if err != nil {
-		return err
+	if c.channel != nil {
+		err = c.validateKubernetesVersion()
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
in addition to nil, c.channel can also point to an empty `Channel{}`:
 https://github.com/kubernetes/kops/blob/781279e268847579b47a87600380675317233b58/pkg/apis/kops/channel.go#L88

maybe we should handle that case as well?

Reply:
Looking at the code for validateKubernetesVersion, it does seem like (1) there are some extra checks in validateKubernetesVersion and (2) validateKubernetesVersion does tolerate the idea that the kubernetes version might not be listed in the channel.

So I suggest we tolerate a nil channel in validateKubernetesVersion, as we do empty kubernetesVersions, but we should always call into validateKubernetesVersion.

Reply:
Returning nil if the channel is nil just before the version recommendation/requirements checks now